grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -45,21 +48,98 @@ class CycleCountApiController {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startCycleCount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
+            xls {
+                exportCountXls(cycleCounts)
+            }
+            pdf {
+                renderCountPdf(cycleCounts, command.facility.name)
+            }
+            json {
+                render([data: cycleCounts] as JSON)
+            }
+        }
     }
 
     def startRecount(CycleCountStartRecountBatchCommand command) {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startRecount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
+            xls {
+                exportRecountXls(cycleCounts)
+            }
+            pdf {
+                renderRecountPdf(cycleCounts, command.facility.name)
+            }
+            json {
+                render([data: cycleCounts] as JSON)
+            }
+        }
     }
 
     def list() {
         List<String> ids = params.list("id")
         List<CycleCountDto> cycleCounts = cycleCountService.getCycleCounts(ids)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        boolean isRecount = cycleCounts?.any { it.status == CycleCountStatus.INVESTIGATING.name() }
+
+        withFormat {
+            xls {
+                isRecount ? exportRecountXls(cycleCounts) : exportCountXls(cycleCounts)
+            }
+            pdf {
+                String facilityName = cycleCounts?.first()?.cycleCountItems?.first()?.facility?.name  ?: ""
+                isRecount ? renderRecountPdf(cycleCounts, facilityName) : renderCountPdf(cycleCounts, facilityName)
+            }
+            json {
+                render([data: cycleCounts] as JSON)
+            }
+        }
+    }
+
+    def exportCountXls(List<CycleCountDto> cycleCounts) {
+        List<Map> data = cycleCountService.getCountFormXls(cycleCounts)
+        response.setHeader("Content-disposition", "attachment; filename=Count form.xls")
+        response.contentType = "application/vnd.ms-excel"
+        documentService.generateExcel(response.outputStream, data)
+        response.outputStream.flush()
+    }
+
+    def renderCountPdf(List<CycleCountDto> cycleCounts, String facilityName) {
+        renderPdf(
+                template: "/cycleCount/printCount",
+                model: [cycleCounts: cycleCounts, facilityName: facilityName, datePrinted: new Date()],
+                filename: "Count form.pdf"
```

Comment:
do we usually use spaces in the file name? I usually do something like count-form.pdf or count_form.pdf

Reply:
I just gave the naming of the files that were provided as templates in the tickets

Reply:
Added few more details there